WordPress/gutenberg
# ESLint: Fix a bunch of ESLint alignment warnings

## What?
This PR fixes a bunch of ESLint warnings, as well as a couple of TS errors that were unveiled with those fixes.

## Why?
Ideally, we shouldn't have any ESLint warnings/errors. This fixes 151 of the total 167 existing ones.

## How?
We ran `npm run lint:js:fix` and did a couple of TS massaging changes. Basically `EntityRecordKey` is a `number | string`, which required the additional TS changes.

## Testing Instructions
* Verify there are only 16 `jest/expect-expect` or `jest/no-disabled-tests` warnings when running `npm run lint:js`.
* Verify all checks are green.

### Testing Instructions for Keyboard
None

## Screenshots or screencast 
None
